docs/openlaw-object/README.md

```diff
@@ -0,0 +1,2141 @@
+
+```scala
+getStructureFieldDefinitions(
+  variable: VariableDefinition,
+  executionResult: TemplateExecutionResult
+): js.Array[VariableDefinition]
+```
+
+**Parameters**
+
+| Name | Type | Description |
+| ---- | ---- | ----------- |
+| `variable` | [`VariableDefinition`](#variabledefinition) | **Required.** A [`VariableDefinition` object](#variabledefinition) representing the variable. |
+| `executionResult` | [`TemplateExecutionResult`](#templateexecutionresult) | **Required.** The nested object returned from the [`execute` method](#execute) and [`executeForReview` method](#executeforreview). |
+
+Example
+
+```js
+// see examples above for #execute and #executeForReview for parameters
+const executionResult = Openlaw.execute(compiledTemplate.compiledTemplate, {}, params);
+const allVariables = Openlaw.getVariables(executionResult.executionResult, {});
+const variable = allVariables[0]; // first variable in template is a Structure type
+Openlaw.getStructureFieldDefinitions(variable, executionResult.executionResult);
+```
+
+**Response**
+
+Returns an array of [`VariableDefinition` objects](#variabledefinition), each of which includes information about a variable field in the [Structure type](/markup-language/#structure), including name and variable type.
+
+### getStructureFieldValue
+
+Get value of a variable field in a [Structure type](/markup-language/#structure) variable.
+
+```scala
+getStructureFieldValue(
+  variable: VariableDefinition,
+  field: VariableDefinition,
+  structureValue: js.UndefOr[String],
+  executionResult: TemplateExecutionResult
+): js.UndefOr[String]
+```
+
+**Parameters**
+
+| Name | Type | Description |
+| ---- | ---- | ----------- |
+| `variable` | [`VariableDefinition`](#variabledefinition) | **Required.** A [`VariableDefinition` object](#variabledefinition) representing the [Structure type](/markup-language/#structure) variable. |
+| `field` | [`VariableDefinition`](#variabledefinition) | **Required.** A [`VariableDefinition` object](#variabledefinition) representing the variable field. |
+| `structureValue` | `String` | **Required.** The variable fields with input values as a string in JSON format. |
+| `executionResult` | [`TemplateExecutionResult`](#templateexecutionresult) | **Required.** The nested object returned from the [`execute` method](#execute) and [`executeForReview` method](#executeforreview). |
+
+Example
+
+```js
+// see examples above for #execute and #executeForReview for parameters
+const executionResult = Openlaw.execute(compiledTemplate.compiledTemplate, {}, params);
+const allVariables = Openlaw.getVariables(executionResult.executionResult, {});
+const variable = allVariables[0]; // first variable in template is a Structure type
+const field = Openlaw.getStructureFieldDefinitions(variable, executionResult.executionResult)[0]; // variable being checked is first variable field in Structure type variable
+const structureValue = '{"First name":"John","Last 
```

Comment:
I think it could be useful to have headings for (for example) methods on collections, methods on structure types, etc...

Reply:
@outkaj the category headings already include Template, Variable, Address, Choice, Collection, Identity, Structure, and Shared Types. Are you thinking the headings should be different?

Reply:
@Jdville03 missed that, sorry, I was looking just from text and not from deploy preview.